src/Gaffer/Reference.cpp

```diff
@@ -153,6 +154,15 @@ Reference::~Reference()
 
 void Reference::load( const std::string &fileName )
 {
+	const char *s = getenv( "GAFFER_REFERENCE_PATHS" );
+	IECore::SearchPath sp( s ? s : "" );
+
+	boost::filesystem::path path = sp.find( fileName );
+	if( path.empty() )
+	{
+		throw Exception( "Could not find file '" + fileName + "'" );
```

Comment:
Maybe it's worth changing the message to something along the lines of "Could not find file "%s" in any of GAFFER_REFERENCE_PATHS"? I think a lot of our exceptions have a tendency to be a little too terse and don't give users enough information to debug their problem by themselves.

Reply:
the find() call also checks if the file is [absolute](https://github.com/ImageEngine/cortex/blob/master/src/IECore/SearchPath.cpp#L109) and returns an empty string, so mentioning GAFFER_REFERENCE_PATHS in the message would be misleading to the user if they supplied an invalid absolute path. So in this case "Could not find file" is a more general message that covers all cases.

I could add more code that handles the specific cases - one if branch where file.is_complete() is true and find() is empty and another where file.is_complete() is false (the search path case) and find() is empty.

Just thought .. there is also the case where GAFFER_REFERENCE_PATHS is not set, should we mention GAFFER_REFERENCE_PATHS even if it is not set?.. once you start addressing corner cases it is a slippery slope to address them all!

Happy to make changes if there is a strong feeling all these specific cases should be detected and specific messages thrown.

Thanks

Reply:
I hadn't looked at the code too much and didn't realize the intricacies. I don't feel strongly about it. Thanks for clarifying!